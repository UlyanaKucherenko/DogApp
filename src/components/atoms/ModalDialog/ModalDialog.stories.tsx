import { action } from '@storybook/addon-actions';
import ModalDialog from './index';

export default {
  title: 'Atoms/ModalDialog',
  component: ModalDialog
};

export const Default = () => (
  <ModalDialog open onClose={action('Close button clicked')} />
);

export const Closed = () => (
  <ModalDialog open={false} onClose={action('Close button clicked')} />
);
