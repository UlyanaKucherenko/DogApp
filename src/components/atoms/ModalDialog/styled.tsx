import { Dialog, DialogTitle, IconButton } from '@mui/material';
import { styled } from '@mui/material/styles';

const StyledDialog = styled(Dialog)({
  display: 'flex',
  alignItems: 'center',
  justifyContent: 'center'
});

const StyledDialogTitle = styled(DialogTitle)({
  fontSize: '40px'
});

const CloseBtn = styled(IconButton)(() => ({
  position: 'absolute',
  top: '10px',
  right: '10px'
}));

export { StyledDialog, StyledDialogTitle, CloseBtn };
