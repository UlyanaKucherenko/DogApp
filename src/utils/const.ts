export const menuHeader = [
  {
    title: 'Main',
    route: '/'
  },
  {
    title: 'Breeds',
    route: '/breeds'
  },
  {
    title: 'Vote',
    route: '/vote'
  },
  {
    title: 'History',
    route: '/history'
  },
  {
    title: 'Your uploads',
    route: '/your-uploads'
  }
];
