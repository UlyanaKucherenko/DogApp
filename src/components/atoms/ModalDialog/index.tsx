import {
  Box,
  Button,
  DialogActions,
  DialogContentText,
  DialogContent
} from '@mui/material';

import CloseIcon from '../../../assets/icons/CloseIcon.svg?react';
import { CloseBtn, StyledDialog, StyledDialogTitle } from './styled';

interface ModalDialogProps {
  open: boolean;
  onClose?: () => void;
  confirm?: () => void;
  title?: string;
  description?: string;
  textBtnClose?: string;
  textBtnConfirm?: string;
  children?: React.ReactNode;
}

const ModalDialog = ({
  open,
  onClose,
  confirm,
  title = 'Upload a Dog image',
  description = 'Information for this dialog',
  textBtnClose = 'close',
  textBtnConfirm = 'confirm',
  children
}: ModalDialogProps) => (
  <StyledDialog
    open={open}
    onClose={onClose}
    PaperProps={{
      style: {
        borderRadius: 20,
        background: 'white',
        boxShadow: '6px 6px 0px 0px #000000',
        maxWidth: '100%'
      }
    }}
  >
    <CloseBtn onClick={onClose}>
      <CloseIcon />
    </CloseBtn>
    {children ? (
      <Box>{children}</Box>
    ) : (
      <Box sx={{ padding: 1 }}>
        <StyledDialogTitle>{title}</StyledDialogTitle>
        <DialogContent>
          <DialogContentText>{description}</DialogContentText>
        </DialogContent>
        <DialogActions sx={{ justifyContent: 'center' }}>
          <Button onClick={onClose} color="primary" variant="outlined">
            {textBtnClose}
          </Button>
          <Button onClick={confirm} color="primary" variant="contained">
            {textBtnConfirm}
          </Button>
        </DialogActions>
      </Box>
    )}
  </StyledDialog>
);
export default ModalDialog;
