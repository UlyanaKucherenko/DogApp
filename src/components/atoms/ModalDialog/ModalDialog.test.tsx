// import { render, fireEvent, screen } from '@testing-library/react';
// import ModalDialog from './index';
// import '@testing-library/jest-dom/';

// describe('ModalDialog component', () => {
//   it('renders with a title', () => {
//     render(<ModalDialog open onClose={() => {}} />);
//     const titleElement = screen.getByText('Upload a Dog image');
//     expect(titleElement).toBeInTheDocument();
//   });

//   it('closes when the Close button is clicked', () => {
//     const handleClose = jest.fn();
//     render(<ModalDialog open onClose={handleClose} />);
//     const closeButton = screen.getByText('Close');
//     expect(handleClose).not.toHaveBeenCalled();
//     fireEvent.click(closeButton);
//     expect(handleClose).toHaveBeenCalled();
//   });
// });

import { composeStories } from '@storybook/react';
import * as stories from './ModalDialog.stories';
import { testSnapshot } from '../../../test/utils/snapshot.utils';

describe('atoms/ModalDialog', () => {
  const { Default } = composeStories(stories);
  testSnapshot(<Default />);
});
