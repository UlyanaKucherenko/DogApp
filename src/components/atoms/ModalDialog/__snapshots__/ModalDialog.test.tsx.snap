// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`atoms/ModalDialog should match snapshot Component 1`] = `
<div
  aria-hidden="true"
/>
`;
